cpp/box/Box.h

```diff
@@ -295,10 +295,25 @@ class Box
      */
     vec3<float> wrap(const vec3<float>& v) const
     {
+        // Return quickly if the box is aperiodic
+        if (!m_periodic.x && !m_periodic.y && !m_periodic.z)
+        {
+            return v;
+        }
+
         vec3<float> v_frac = makeFractional(v);
-        v_frac.x = util::modulusPositive(v_frac.x, 1.0f);
-        v_frac.y = util::modulusPositive(v_frac.y, 1.0f);
-        v_frac.z = util::modulusPositive(v_frac.z, 1.0f);
+        if (m_periodic.x)
+        {
+            v_frac.x = util::modulusPositive(v_frac.x, 1.0f);
+        }
+        if (m_periodic.y)
+        {
+            v_frac.y = util::modulusPositive(v_frac.y, 1.0f);
+        }
+        if (m_periodic.z)
+        {
+            v_frac.z = util::modulusPositive(v_frac.z, 1.0f);
+        }
```

Comment:
I'm unfamiliar with many of the calculations in freud right now, did you have any in mind specifically that could benefit from this type of error message?

Reply:
Offhand I'd say anything that touches neighbor finding, you wouldn't want to find periodic neighbors in an aperiodic box. So I would probably put errors in the `query` method of `NeighborQuery` (you might have to do it in the subclasses, I honestly can't remember if I made that method is pure virtual or not). I _think_ that all non-pair computations other than `GaussianDensity` (and now `SphereVoxelization`) should be safe because they just interact with the point positions or orientations. These two are special because the smearing requires accounting for periodicity, but I don't think any of the others do.

Reply:
`query()` is not pure virtual, so I'll add an error there. That should be the only place where we have not already accounted for box periodicity right?